src/gui/sdlmain.cpp

```diff
@@ -3478,19 +3504,48 @@ static void GUI_StartUp(Section *sec)
 	startup_state_capslock = keystate & KMOD_CAPS;
 }
 
-static void HandleMouseMotion(SDL_MouseMotionEvent * motion) {
-	if (mouse_is_captured || sdl.mouse.control_choice == Seamless)
-		Mouse_CursorMoved((float)motion->xrel*sdl.mouse.xsensitivity/100.0f,
-						  (float)motion->yrel*sdl.mouse.ysensitivity/100.0f,
-						  (float)(motion->x-sdl.clip.x)/(sdl.clip.w-1)*sdl.mouse.xsensitivity/100.0f,
-						  (float)(motion->y-sdl.clip.y)/(sdl.clip.h-1)*sdl.mouse.ysensitivity/100.0f,
-						  mouse_is_captured);
+
+static void HandleMouseMotion(SDL_MouseMotionEvent *motion)
+{
+	if (motion->xrel < INT16_MIN || motion->xrel > INT16_MAX ||
+	    motion->yrel < INT16_MIN || motion->yrel > INT16_MAX || motion->x < 0 ||
+	    motion->x > UINT16_MAX || motion->y < 0 || motion->y > UINT16_MAX) {
+		LOG_ERR("SDL: Unexpected motion values: rel %dx%d, abs %dx%d",
+		        motion->xrel,
+		        motion->xrel,
+		        motion->x,
+		        motion->y);
+		return;
+	}
+
+	if (motion->xrel != 0 || motion->yrel != 0) {
+		if (mouse_vmware || mouse_is_captured ||
+		    sdl.mouse.control_choice == Seamless)
+			MOUSE_EventMoved(static_cast<int16_t>(motion->xrel),
+			                 static_cast<int16_t>(motion->yrel),
+			                 static_cast<uint16_t>(motion->x),
+			                 static_cast<uint16_t>(motion->y),
```

Comment:
Seeing these casts at the end-point where the values are used, tells me we should go upstream and correct the types inside the `motion` object.  If these should be 16-bit values, then let's make their types `int16_t` and `uint16_t` from the get-go, and avoid casting here.

Reply:
Upstream won't like the idea of breaking the compatibility, they won't accept this for SDL 2 for sure. And current 8K screens are 7680x4320; above 32K resolution things start getting funny with 16-bit variables :)

Reply:
Yeah, these are small but tricky decisions.  

If it makes sense to hold the coordinates in  full-resolution, "host-side" variables, saying, SDL "int", then that's fine too.  And then only scale down to 16-bit at the DOS API boundaries.

But if those coordinates get sloshed back and fourth across the 16-bit API, then they'll be truncated for good anyway, in which case, I'd go with the DOS 16-bit type, and covert the SDL values down at the point where you can retain the most resolution. 

I don't know much about the mouse code or APIs though ; please run with what you feel is best.

For some flexibility, feel free to make a mouse-specific coordinate type:

**mouse.h**
``` c++
using host_mouse_pos_t = int16_t;
using dos_mouse_pos_t = int16_t;

```

Then when everyone's playing DOSBox on their 48,000 x 72,000 res smart watches, we can boost up `host_mouse_pos_t = int` :sweat_smile:

Reply:
In the mixer, I had a similar dilema regarding the worker-buffer indexes (and if we ever wanted to go above a 64 KB buffer), so use `work_index_t` for all the offsets:

``` c++
using work_index_t = uint16_t;
```

Reply:
> Upstream won't like the idea of breaking the compatibility, they won't accept this for SDL 2 for sure. 

As long as we do our best for _"the here-and-now"_ given the balance of language features (Circa C++17) and SDL and DOS APIs on-hand, then that's as good as we can do.   Future maintainers, back-porters, and AI-code-generators can do the same if and when that day comes. :+1: